script.js

```diff
@@ -10,7 +10,7 @@ document.getElementsByTagName('pre')[0] !== undefined) {
         const parsed = parse(text)
         const rendered = render(parsed)
 
-        const title = parsed.find(element => element.type === 'header').content || 'Gemini Document' // Find relevant page content here?
+        const title = parsed.find(({ type }) => type === 'header').content || location.href 
```

Comment:
This will throw an error if there are no `type: header` tokens because it will try to get `content` of `undefined`.

Should probably do two steps: get the token, if it exists get the content for the title, if it doesn't use the location href.

Reply:
Ah of course! There had to have been a reason you were doing it the way you were...
I'll fix that and add the fix for if there is interference from other extensions

Reply:
Fixed